xamarin/Xamarin.Forms
# [iOS,Android, UWP] Set Position and CurrentItem when ItemsSource items change

### Description of Change ###

When we add or remove items from the ItemsSource, we need to tell the new Position/CurrentItem on the Carousel. This change will listen to changes on Observable Collections and get the new position, when we set the Position we update the CurrrentItem.

Also resets the ItemDecoration on android when adding or removing items in the last position, since we add an extra spacing to those items so they can be centered. (#9401)

Fixes a small issue with IndicatorView on UWP when it's disposing.

Adds 2 new properties to have more fine grain control over the `Position` and `CurrentItem` updates. If a user want's to just jump to a position when setting the CarouselView Position property  without animation, he can override AnimatePositionChanges and set it to false, while animate CurrentItems changes or vice versa. By default this properties are hooked tot IsScrollAnimated

### Issues Resolved ### 

- fixes #9322
- fixes #9401 
- fixes #9432 
- fixes #9771

### API Changes ###

Removed from CarouseView.cs

- `public void SetCurrentItem(object item)`
- `public void PlatformInitialized()`
- `public void PlatformInitialized()`
- `public Queue<Action> ScrollToActions = new Queue<Action>();`

Added to CarouseView.cs
```
	[EditorBrowsable(EditorBrowsableState.Never)]
	public virtual bool AnimatePositionChanges => IsScrollAnimated;

	[EditorBrowsable(EditorBrowsableState.Never)]
	public virtual bool AnimateCurrentItemChanges => IsScrollAnimated;
```
### Platforms Affected ### 

- iOS
- Android

### Behavioral/Visual Changes ###


Should update CurrentItem

### Before/After Screenshots ### 


Not applicable

### Testing Procedure ###

Enable the Indicator Flag.

Go to CarouselXamlGallery, swipe to the second item , click on the button to Remove make sure the Position and the CurrentItem change.

To test 9432 go to IndicatorCodeGallery , swipe to the last item, click remove, check that the Indicators update and the last one is still selected.

To test 9401 , visit the CarouselView / CarouselView (Indicators Forms), go to the last item, click add item,  make sure the added item doesn't show a extra spacing on the left of the last item on Android.

Test 9771 issue page, make sure you go to 50 and can go to previous and next on UWP

### PR Checklist ###
